mqtt/mqtt-bridge/src/controller.rs

```diff
@@ -1,47 +1,42 @@
-use std::collections::HashMap;
-
 use futures_util::future::{self, join_all};
+use tokio::task;
 use tracing::{error, info};
 
-use crate::bridge::{Bridge, BridgeError};
-use crate::settings::Settings;
+use crate::bridge::Bridge;
+use crate::settings::BridgeSettings;
 
 /// Controller that handles the settings and monitors changes, spawns new Bridges and monitors shutdown signal.
-#[derive(Default)]
-pub struct BridgeController {
-    bridges: HashMap<String, Bridge>,
-}
+pub struct BridgeController {}
 
 impl BridgeController {
-    pub fn new() -> Self {
-        Self::default()
-    }
+    pub async fn run(system_address: String, device_id: String, settings: BridgeSettings) {
+        info!("starting bridge controller...");
 
-    pub async fn init(
-        &mut self,
-        system_address: String,
-        device_id: &str,
-    ) -> Result<(), BridgeError> {
-        info!("initializing bridge controller...");
-        let settings = Settings::new().map_err(BridgeError::LoadingSettings)?;
-
-        if let Some(upstream) = settings.upstream() {
-            let bridge = Bridge::new(system_address, device_id.into(), upstream.clone());
-            self.bridges.insert(upstream.name().to_string(), bridge);
+        let mut bridge_handles = vec![];
```

Comment:
hashmap will be useful for configuration from twin, at that point the bridge will be searched by name

Reply:
The issue with this is that the bridge cannot be shared across threads without using locks. We can talk about the best approach for a workaround.